gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/exception/DefaultPubSubMessageReceiverExceptionHandler.java

```diff
@@ -30,6 +31,7 @@
  * @since 2.0.0
  */
 @Singleton
+@Primary
```

Comment:
I don't think this is correct. It is smelly we have to force our bean to be `@Primary`

Reply:
@sdelamo 

> I don't think this is correct. It is smelly we have to force our bean to be `@Primary`

I annotated it as `@Primary` because that's the way the [default Kafka exception handler](https://github.com/micronaut-projects/micronaut-kafka/blob/master/kafka/src/main/java/io/micronaut/configuration/kafka/exceptions/DefaultKafkaListenerExceptionHandler.java#L40) was implemented in the Kafka module.

> Have you tried annotating it with `@Secondary`?

Yes I have; it doesn't work when it's annotated as `@Secondary`.

Reply:
This still looks smelly to me, but it seems we created the API with the @Primary in mind. The test which shows how to override the default handler also uses @Primary https://github.com/micronaut-projects/micronaut-gcp/blob/master/gcp-pubsub/src/test/groovy/io/micronaut/gcp/pubsub/bind/ExceptionHandlerSpec.groovy#L104

I am going to merge this and hopefully we can have a better design in the future.

Reply:
The design is a bit messy, but I guess the primary is OK here